types/src/lib.rs

```diff
@@ -97,7 +97,3 @@ pub use crate::{
 compile_error!(
     "casper-types requires one of 'std' (enabled by default) or 'no-std' features to be enabled"
 );
-#[cfg(all(feature = "std", feature = "no-std"))]
```

Comment:
Removing this seems incorrect.  The `std` and `no-std` features in this crate are meant to be mutually exclusive and due to the limitations of Rust manifests, this is the only way that is enforced now.

It has made me notice that the error message for this and the `compile_error!` above are both wrong - `no-std` is enabled by default.

Reply:
This has been breaking several people's tooling (ie, `rust-analyzer`).  We are going to have to figure out something here that works for everyone.

Reply:
I only know of one person's tooling.  However, even if it's affecting more, let's not make an unrelated change like that in this PR please?

Reply:
Okay, so chatting this over with @Fraser999 , the concern here is that smart-contract writers might get knackered if they have both `std` and `no-std` feature flags set.

Reply:
Reverted in d32751566c18a5b587030abbf4f664e740c75b54